client/src/components/AddList/AddList.tsx

```diff
@@ -0,0 +1,86 @@
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Box from '@material-ui/core/Box';
+import { Formik } from 'formik';
+import Typography from '@material-ui/core/Typography';
+import useStyles from './useStyles';
+import { CircularProgress } from '@material-ui/core';
+import Container from '@material-ui/core/Container';
+import { Icon, DialogActions, Grid, CssBaseline, Paper, IconButton } from '@material-ui/core';
+import { addNewList } from '../../helpers/APICalls/lists';
+
+interface Props {
+  handleSubmit: () => void;
+}
+
+export default function AddList({ onClose }: any): JSX.Element {
+  const classes = useStyles();
+
+  const handleSubmit = (data: any) => {
+    addNewList(data);
+    onClose();
+  };
+
+  const renderModalContent = () => {
+    return (
+      <Container component="main" style={{ minWidth: '670px' }} maxWidth="xs" className="bg-secondary">
+        <Formik
+          initialValues={{
+            name: '',
+          }}
+          onSubmit={handleSubmit}
+        >
```

Comment:
Formik has great support for input validation with Yup, I would add some error handling here if they try to submit an empty list name.